client.go

```diff
@@ -415,6 +414,53 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 	return conn, resp, nil
 }
 
+// Returns an updated proxy dial function, wrapping the passed
+// "netDial" function, or an error if one occurred. If the proxy
+// is HTTPS, then TLS handshake will occur in the returned dial function.
+func (d *Dialer) proxyNetDial(req *http.Request, netDial netDialerFunc) (netDialerFunc, error) {
+	if d.Proxy == nil {
+		return nil, errors.New("proxy url function is nil")
+	}
+	// Generate the proxy URL and validate.
+	proxyURL, err := d.Proxy(req)
+	if err != nil {
+		return nil, err
+	}
+	if proxyURL == nil {
+		return nil, errors.New("generated proxy url is nil")
+	}
+	if proxyURL.Scheme == "http" && d.NetDialTLSContext != nil {
+		return nil, errors.New("dial TLS function set for dialing non-HTTPS proxy")
+	}
```

Comment:
This does not match the logic in 
```
	var netDial netDialerFunc
	switch {
	case u.Scheme == "https" && d.NetDialTLSContext != nil:
		netDial = d.NetDialTLSContext
	case d.NetDialContext != nil:
		netDial = d.NetDialContext
	case d.NetDial != nil:
		netDial = func(ctx context.Context, net, addr string) (net.Conn, error) {
			return d.NetDial(net, addr)
		}
	default:
		netDial = (&net.Dialer{}).DialContext
	}
```

that seems to indicate that `NetDialTLSContext` is ONLY used if the scheme is https , but ignored otherwise